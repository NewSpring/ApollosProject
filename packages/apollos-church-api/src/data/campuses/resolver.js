import { Campus as originalCampus } from '@apollosproject/data-connector-rock';
import { schemaMerge } from '@apollosproject/server-core';
import { createAssetUrl } from '../utils';

const resolver = {
  Query: {
    campuses: (root, { location }, { dataSources }) =>
      dataSources.Campus.getPublicByLocation(location),
  },
  Campus: {
    latitude: ({ location }) =>
      (typeof location.latitude === 'number' && location.latitude) || 0,
    longitude: ({ location }) =>
      (typeof location.longitude === 'number' && location.longitude) || 0,
    image: ({ location }) => ({
      uri: createAssetUrl(location.image),
      width: location.image.width,
      height: location.image.height,
    }),
  },
};

export default schemaMerge(resolver, originalCampus);
