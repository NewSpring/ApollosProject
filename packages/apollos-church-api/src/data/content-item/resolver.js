import { ContentItem as originalContentItem } from '@apollosproject/data-connector-rock';
import { schemaMerge } from '@apollosproject/server-core';
// import { get } from 'lodash';

import ApollosConfig from '@apollosproject/config';

const { ROCK_MAPPINGS } = ApollosConfig;

const resolver = {
  DevotionalContentItem: {
    scriptures: async ({ id }, args, { dataSources }) => {
      const scriptures = await dataSources.ContentItem.getContentItemScriptures(
        id
      );
      if (scriptures.length) {
        return dataSources.Scripture.getScriptures(scriptures);
      }
      return [];
    },
  },
  ContentItem: {
    __resolveType: async (
      { attributeValues, attributes, contentChannelId },
      { dataSources: { ContentItem } }
    ) => {
      // if we have defined an ContentChannelId based maping in the YML file, use it!
      if (
        Object.values(ROCK_MAPPINGS.CONTENT_ITEM).some(
          ({ ContentChannelId }) =>
            ContentChannelId && ContentChannelId.includes(contentChannelId)
        )
      ) {
        return Object.keys(ROCK_MAPPINGS.CONTENT_ITEM).find((key) => {
          const value = ROCK_MAPPINGS.CONTENT_ITEM[key];
          return (
            value.ContentChannelId &&
            value.ContentChannelId.includes(contentChannelId)
          );
        });
      }

      if (ContentItem.hasMedia({ attributeValues, attributes })) {
        return 'MediaContentItem';
      }

      return 'UniversalContentItem';
    },
  },
};

export default schemaMerge(resolver, originalContentItem);
