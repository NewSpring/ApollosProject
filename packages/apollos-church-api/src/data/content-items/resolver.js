import { ContentItem as originalContentItem } from '@apollosproject/data-connector-rock';
import { resolverMerge } from '@apollosproject/server-core';

import ApollosConfig from '@apollosproject/config';

const { ROCK_MAPPINGS } = ApollosConfig;

// TODO temp fix, remove after this is merged
// https://github.com/ApollosProject/apollos-prototype/pull/1061
const defaultResolvers = {
  sharing: (root, args, { dataSources: { ContentItem } }) => ({
    url: ContentItem.getShareUrl(root.id, root.contentChannelId),
    title: 'Share via ...',
    message: `${root.title} - ${ContentItem.createSummary(root)}`,
  }),
};

const resolver = {
  Query: {
    contentItemFromSlug: (root, { slug }, { dataSources }) =>
      dataSources.ContentItem.getBySlug(slug),
  },
  DevotionalContentItem: {
    ...defaultResolvers,
    scriptures: async (
      { attributeValues: { scriptures } = {} },
      args,
      { dataSources }
    ) => dataSources.ContentItem.getContentItemScriptures(scriptures),
    series: ({ id }, args, { dataSources: { ContentItem } }) =>
      ContentItem.getParent(id, ROCK_MAPPINGS.DEVOTIONAL_SERIES_CHANNEL_ID),
  },
  ContentItem: {
    __resolveType: async (
      { attributeValues, attributes, contentChannelId },
      { dataSources: { ContentItem } }
    ) => {
      // if we have defined an ContentChannelId based maping in the YML file, use it!
      if (
        Object.values(ROCK_MAPPINGS.CONTENT_ITEM).some(
          ({ ContentChannelId }) =>
            ContentChannelId && ContentChannelId.includes(contentChannelId)
        )
      ) {
        return Object.keys(ROCK_MAPPINGS.CONTENT_ITEM).find((key) => {
          const value = ROCK_MAPPINGS.CONTENT_ITEM[key];
          return (
            value.ContentChannelId &&
            value.ContentChannelId.includes(contentChannelId)
          );
        });
      }

      if (ContentItem.hasMedia({ attributeValues, attributes })) {
        return 'MediaContentItem';
      }

      return 'UniversalContentItem';
    },
  },
  // SharableContentItem: {
  // url: ({ id, contentChannelId }, args, { dataSources }) =>
  // dataSources.ContentItem.getShareUrl(id, contentChannelId),
  // },
  WeekendContentItem: {
    ...defaultResolvers,
    communicator: (
      { attributeValues: { communicators } = {} },
      args,
      { dataSources }
    ) => dataSources.ContentItem.getCommunicator(communicators),
    sermonDate: ({ attributeValues: { actualDate: { value } = {} } = {} }) =>
      value,
    series: ({ id }, args, { dataSources: { ContentItem } }) =>
      ContentItem.getParent(id, ROCK_MAPPINGS.SERMON_SERIES_CHANNEL_ID),
  },
  UniversalContentItem: {
    ...defaultResolvers,
  },
  MediaContentItem: {
    ...defaultResolvers,
  },
};

export default resolverMerge(resolver, originalContentItem);
