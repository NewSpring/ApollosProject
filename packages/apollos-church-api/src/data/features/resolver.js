import { Features as baseFeatures } from '@apollosproject/data-connector-rock';
import { resolverMerge } from '@apollosproject/server-core';

const resolver = {
  HeaderFeature: {
    // TODO waiting on core release
    // sharing: ({ body }) => ({
    // title: 'Share via...',
    // message: body,
    // }),
  },
};

export default resolverMerge(resolver, baseFeatures);
