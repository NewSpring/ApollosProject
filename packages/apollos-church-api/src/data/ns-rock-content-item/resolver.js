import { ContentItem } from '@apollosproject/data-connector-rock';
import { schemaMerge } from '@apollosproject/server-core';

const resolver = {
  NewSpringContentItem: {
    ...ContentItem.resolver.ContentItem,
  },
};

export default schemaMerge(resolver, ContentItem);
