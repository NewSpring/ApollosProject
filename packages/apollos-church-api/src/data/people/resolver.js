import { Person as originalPerson } from '@apollosproject/data-connector-rock';
import { resolverMerge } from '@apollosproject/server-core';

const resolver = {
  Person: {
    // deprecated
    impersonationParameter: () => 'deprecatedtoken',
    isGroupLeader: async ({ id }, args, { dataSources: { Group } }) => {
      const groups = await Group.getByPerson({ personId: id, asLeader: true });
      return groups.length > 0;
    },
    isStaff: ({ id }, args, { dataSources: { Person } }) => Person.isStaff(id),
    photo: ({ photo: { path } }) => ({ uri: path }),
  },
};
export default resolverMerge(resolver, originalPerson);
