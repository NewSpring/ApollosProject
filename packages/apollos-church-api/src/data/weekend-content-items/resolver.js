import { ContentItem as baseContentItem } from '@apollosproject/data-connector-rock';
import { resolverMerge } from '@apollosproject/server-core';

const resolver = {
  WeekendContentItem: {
    communicator: () => null,
    sermonDate: ({ attributeValues: { actualDate: { value } = {} } = {} }) =>
      value,
  },
};
export default resolverMerge(resolver, baseContentItem);
