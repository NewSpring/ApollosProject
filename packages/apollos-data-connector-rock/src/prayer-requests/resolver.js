import { createGlobalId } from '@apollosproject/server-core';

export default {
  Query: {
    getPrayerRequests: (root, args, { dataSources }) =>
      dataSources.PrayerRequest.getPrayerRequests(),
  },
  //   Mutation: {
  //     updateProfileField: (root, { input: { field, value } }, { dataSources }) =>
  //       dataSources.Person.updateProfile([{ field, value }]),
  //     updateProfileFields: (root, { input }, { dataSources }) =>
  //       dataSources.Person.updateProfile(input),
  //     uploadProfileImage: async (root, { file, size }, { dataSources }) =>
  //       dataSources.Person.uploadProfileImage(file, size),
  //   },
  PrayerRequest: {
    id: ({ id }, args, context, { parentType }) =>
      createGlobalId(id, parentType.name),
  },
};
