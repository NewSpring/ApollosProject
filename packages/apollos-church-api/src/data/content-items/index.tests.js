import resolver from '.';

describe('a test', () => {
  it('must make a snapshot', () => {
    expect(resolver).toMatchSnapshot();
  });
});
