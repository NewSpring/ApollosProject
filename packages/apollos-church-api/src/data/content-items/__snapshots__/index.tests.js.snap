// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a test must make a snapshot 1`] = `
Object {
  "DevotionalContentItem": Object {
    "theme": [Function],
  },
  "MediaContentItem": Object {
    "theme": [Function],
    "videos": [Function],
  },
  "UniversalContentItem": Object {
    "theme": [Function],
  },
}
`;
